### PR TITLE
Snake example

### PR DESCRIPTION
**Pull request by svenwiltink**

### CODE REVIEW

File: examples/snake/snake.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+	w = 500
+	h = 500
+
+	blockSize = 10
+)
+
+func main() {
+	rand.Seed(time.Now().UnixNano())
+	snake := SnakeGame{}
+	game := ardent.NewGame("Snake", w, h, engine.FlagResizable, snake.Tick, func(_ int, _ int) (int, int) {
```
Comment (zacharyburkett) [inline examples/snake/snake.go:23]: Nitpick, but maybe spread this along multiple lines?

File: examples/snake/snake.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+		return w, h
+	})
+
+	snake.setup(game)
+
+	err := game.Run()
```
Comment (zacharyburkett) [inline examples/snake/snake.go:29]: Also nitpick, but this could be inline with the if statement

File: examples/snake/snake.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+	lastMove    time.Time
+	timePerMove time.Duration
+
+	lastInput int
+}
+
+func (g *SnakeGame) Tick() {
+	g.checkInput()
+
+	now := time.Now()
```
Comment (zacharyburkett) [inline examples/snake/snake.go:59]: Doesn’t look like we need the variable, we could just call `time.Now()` below

File: examples/snake/snake.go [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+		img: img,
+	}
+
+	b.Translate(x, y)
+	g.renderer.AddImage(img)
+	return &b
+}
+
+// DrawBorder creates the game border using four individual images. This could be one image as well.
+func (g *SnakeGame) DrawBorder() {
```
Comment (zacharyburkett) [inline examples/snake/snake.go:117]: Could we maybe simplify this to use an image? Just to reduce the code complexity of the examples. Don’t want any hur dur snake is only X lines in engine Y

**Review by zacharyburkett — CHANGES_REQUESTED** [review]
Overall I just have some nitpicky change requests, but very solid! 🔥

File: examples/snake/snake.go (hunk 4 above)
Comment (svenwiltink) [reply examples/snake/snake.go:119]: Not entirely sure what you mean. You mean drawing the border on a single `draw.Image` or just creating a background image and using that for the game?

**Review by zacharyburkett — APPROVED** [review]

**Pull request merged**